### PR TITLE
Salesforce docs & remove unnecessary fields

### PR DESCRIPTION
**Pull request by ulixius9**
### Describe your changes:

Salesforce docs & remove unnecessary fields

I worked on ... because ...

#
### Type of change:

- [ ] Bug fix
- [ ] Improvement
- [ ] New feature
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation

#
### Checklist:

- [x] I have read the [**CONTRIBUTING**](https://docs.open-metadata.org/developers/contribute) document.
- [ ] My PR title is `Fixes <issue-number>: <short explanation>`
- [ ] I have commented on my code, particularly in hard-to-understand areas. 
- [ ] For JSON Schema changes: I updated the migration scripts or explained why it is not needed.

### CODE REVIEW

File: openmetadata-spec/src/main/resources/json/schema/entity/services/connections/database/salesforceConnection.json [hunk 1]
```diff
@@ -26,12 +26,6 @@
       "$ref": "#/definitions/salesforceType",
       "default": "Salesforce"
     },
-    "scheme": {
```
Comment (ulixius9) [inline openmetadata-spec/src/main/resources/json/schema/entity/services/connections/database/salesforceConnection.json:29]: salesforce is not an SQA source hence scheme is not required

File: openmetadata-spec/src/main/resources/json/schema/entity/services/connections/database/salesforceConnection.json [hunk 2]
```diff
@@ -49,18 +43,13 @@
       "type": "string",
       "format": "password"
     },
-    "hostPort": {
-      "title": "Host and Port",
```
Comment (ulixius9) [inline openmetadata-spec/src/main/resources/json/schema/entity/services/connections/database/salesforceConnection.json:53]: hostport field is not being used anywhere

File: openmetadata-spec/src/main/resources/json/schema/entity/services/connections/database/salesforceConnection.json [hunk 3]
```diff
@@ -75,14 +64,6 @@
     "supportsMetadataExtraction": {
       "title": "Supports Metadata Extraction",
       "$ref": "../connectionBasicType.json#/definitions/supportsMetadataExtraction"
-    },
-    "supportsProfiler": {
```
Comment (ulixius9) [inline openmetadata-spec/src/main/resources/json/schema/entity/services/connections/database/salesforceConnection.json:79]: salesforce is a non SQA source hence does not support profiled

File: openmetadata-spec/src/main/resources/json/schema/entity/services/connections/database/salesforceConnection.json [hunk 4]
```diff
@@ -75,14 +64,6 @@
...
-      "title": "Supports Profiler",
-      "$ref": "../connectionBasicType.json#/definitions/supportsProfiler"
-    },
-    "supportsQueryComment": {
-      "title": "Supports Query Comment",
```
Comment (ulixius9) [inline openmetadata-spec/src/main/resources/json/schema/entity/services/connections/database/salesforceConnection.json:84]: we do not execute any query to fetch data from salesforce hence this field is useless

**Review by pmbrull — APPROVED** [review]

File: openmetadata-spec/src/main/resources/json/schema/entity/services/connections/database/salesforceConnection.json (hunk 1 above)
Comment (nauxliu) [reply openmetadata-spec/src/main/resources/json/schema/entity/services/connections/database/salesforceConnection.json:29]: Deleting the schema might have caused issue #11468, could you please take a look?

**Pull request merged**